### PR TITLE
chore(codegen): pull dependency versions from package.json

### PR DESCRIPTION
**Pull request by adamthom-amzn**
### Description
When the versions in package.json get out of sync across the generators, we start to get weird build failures in our other test packages. This just pulls the version of the SDK dependencies from each SDK package's package.json

### Testing
I locally generated a client and service with this change and both worked.

### Additional context
This is meant to replace the need for semi-regular updates like https://github.com/aws/aws-sdk-js-v3/commit/597e68980ff9b1cb5daca4ec8eea05c6bce17ff1

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by gosar — CHANGES_REQUESTED** [review]
the couple of packages in that PR that didn’t have 3.15.0 earlier, @aws-sdk/xml-builder, and @aws-sdk/middleware-endpoint-discovery , seem to have 3.18.0 too now. maybe they can also be changed to use LIVE

**Comment by adamthom-amzn** [comment]
> the couple of packages in that PR that didn’t have 3.15.0 earlier, @aws-sdk/xml-builder, and @aws-sdk/middleware-endpoint-discovery , seem to have 3.18.0 too now. maybe they can also be changed to use LIVE

I'm going to solicit the input of people who know why those versions were so widely divergent previously before I make any changes to it.

File: codegen/smithy-aws-typescript-codegen/build.gradle.kts [hunk 1]
```diff
@@ -13,14 +13,44 @@
...
     api("software.amazon.smithy:smithy-aws-cloudformation-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-aws-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-waiters:[1.7.0, 1.8.0[")
     api("software.amazon.smithy:smithy-aws-iam-traits:[1.7.0, 1.8.0[")
     api("software.amazon.smithy.typescript:smithy-typescript-codegen:0.3.0")
 }
+
+tasks.register("set-aws-sdk-version") {
+    doLast {
+        var packageJson = Node.parse(project.file("../../package.json").readText()).expectObjectNode()
```
Comment (trivikr) [inline codegen/smithy-aws-typescript-codegen/build.gradle.kts:41]: The version for individual packages may differ and needs to be read separately.

For example, the package `@aws-sdk/middleware-logger` did not have versions `3.16.0` and `3.17.0` https://github.com/aws/aws-sdk-js-v3/blob/main/packages/middleware-logger/CHANGELOG.md

This PR needs to be updated to read version of individual dependencies.

**Review by trivikr — CHANGES_REQUESTED** [review]

File: codegen/smithy-aws-typescript-codegen/build.gradle.kts (hunk 1 above)
Comment (adamthom-amzn) [reply codegen/smithy-aws-typescript-codegen/build.gradle.kts]: I now walk the top level of both `clients/` and `/packages`, pull the name and version out of every package.json encountered, and bake them into the codegen JAR for reference at runtime.

**Review by gosar — APPROVED** [review]

**Review by trivikr — APPROVED** [review]

**Pull request merged**